src/Lean/Elab/PreDefinition/WF/TerminationHint.lean

```diff
@@ -70,19 +78,31 @@ def TerminationHints.ensureNone (hints : TerminationHints) (reason : String): Co
     logErrorAt hints.ref m!"unused termination hints, function is {reason}"
 
 /--
-Checks that `termination_by` binds at most as many variables are present in the outermost
-lambda of `value`, and logs (without failing) appropriate errors.
+Remembers `extraParams` for later use. Needs to happen early enough where we still know
+how many parameters came from the function header (`headerParams`).
+-/
+def TerminationHints.rememberExtraParams (headerParams : Nat) (hints : TerminationHints)
+    (value : Expr) : TerminationHints :=
+  { hints with extraParams := value.getNumHeadLambdas - headerParams }
 
-Also remembers `extraParams` for later use.
+/--
+Checks that `termination_by` binds at most as many variables are present in the outermost
+lambda of `value`, and throws appropriate errors.
 -/
-def TerminationHints.checkVars (headerParams : Nat) (hints : TerminationHints) (value : Expr) :
-    MetaM TerminationHints := do
-  let extraParams := value.getNumHeadLambdas - headerParams
-  if let .some tb := hints.termination_by? then
+def TerminationBy.checkVars (funName : Name) (extraParams : Nat) (tb : TerminationBy) : MetaM Unit := do
+  unless tb.synthetic do
     if tb.vars.size > extraParams then
-      logErrorAt tb.ref <| m!"Too many extra parameters bound; the function definition only " ++
-        m!"has {extraParams} extra parameters."
-  return { hints with extraParams := extraParams }
+      let mut msg := m!"{parameters tb.vars.size} bound in `termination_by`, but the body of " ++
+        m!"{funName} only binds {parameters extraParams}."
+      if let `($ident:ident) := tb.vars[0]! then
+        if ident.getId.isSuffixOf funName then
+            msg := msg ++ m!"\nNB: Since Lean v4.6, the `termination_by` clause no longer " ++
+              "expects the function name here."
```

Comment:
```suggestion
            msg := msg ++ m!"\nNB: Since Lean v4.6.0, the `termination_by` clause no longer " ++
              "expects the function name here."
```

(Maybe parenthetical, rather than "NB"? Easier to parse, I think.)

Reply:
Also, you can now use multiline strings just by ending with a line with a `\`!

Reply:
Will do.

Too bad we don’t have a “update all expected test output” github action yet…